collectors/modality-probe-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,550 @@
+    } else {
+        // StructOpt will exit if neither are provided
+        unreachable!()
+    };
+
+    Ok(modality_probe_debug_collector::Config {
+        session_id: options.session_id.into(),
+        target,
+        interval,
+        output_path: options.output_path,
+        probe_addrs,
+    })
+}
+
+/// Parse a probe address from a given argument, or return none in case of a symbol
+fn parse_probe_address(input: &str, use_64_bit: bool) -> Result<Option<ProbeAddr>, CLIError> {
+    let is_address = ["0x", "0X", "*0x", "*0X"]
+        .iter()
+        .any(|prefix| input.starts_with(prefix));
+    if !is_address {
+        // Input is a symbol
+        return Ok(None);
+    }
+
+    let trimmed = input
+        .trim_start_matches("*0x")
+        .trim_start_matches("*0X")
+        .trim_start_matches("0x")
+        .trim_start_matches("0X");
+    let addr = u64::from_str_radix(trimmed, 16)
+        .map_err(|_e| CLIError::AddressNotValid(input.to_string()))?;
+    let addr_word = if use_64_bit {
+        Word::U64(addr)
+    } else {
+        let addr_32 =
+            u32::try_from(addr).map_err(|_e| CLIError::AddressNotValid(input.to_string()))?;
+        Word::U32(addr_32)
+    };
+    if input.starts_with('*') {
+        Ok(Some(ProbeAddr::PtrAddr(addr_word)))
+    } else {
+        Ok(Some(ProbeAddr::Addr(addr_word)))
+    }
+}
+
+/// Open elf file for parsing
+fn open_elf<'a>(path: &PathBuf, elf_buf: &'a mut Vec<u8>) -> Result<Elf<'a>, CLIError> {
+    let mut file = File::open(path).map_err(|_e| CLIError::ElfFileError)?;
+    file.read_to_end(elf_buf)
+        .map_err(|_e| CLIError::ElfFileError)?;
+    Elf::parse(elf_buf).map_err(|_e| CLIError::ElfFileError)
+}
+
+/// Get value and size of given symbol, or None if symbol cannot be found
+fn parse_symbol_info(
+    elf_file: &Elf,
+    symbol_name: &str,
+    use_64_bit: bool,
+) -> Result<Word, CLIError> {
+    let log_sym = elf_file
+        .syms
+        .iter()
+        .find(|sym| -> bool {
+            let name_opt = elf_file.strtab.get(sym.st_name);
+            if let Some(Ok(name)) = name_opt {
+                name == symbol_name
+            } else {
+                false
+            }
+        })
+        .ok_or_else(|| CLIError::SymbolNotFound(symbol_name.to_string()))?;
+    if use_64_bit {
+        Ok(Word::U64(log_sym.st_value))
+    } else {
+        let val_32 = u32::try_from(log_sym.st_value)
+            .map_err(|_e| CLIError::AddressNotValid(symbol_name.to_string()))?;
+        Ok(Word::U32(val_32))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::format;
+    use std::fs::canonicalize;
+    use std::process::Command;
+    use std::str::from_utf8;
+    use std::str::FromStr;
+    use std::time::Duration;
+
+    const SYMBOLS_32_BIN_PATH: &str =
+        "./tests/symbols-example/target/thumbv7em-none-eabihf/debug/symbols-example";
+    const EMPTY_BIN_PATH: &str = "./tests/empty-example/target/debug/empty-example";
```

Comment:
As CI noted, this probably ought to be cfg-gated to the targets where it is in use.